language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/CodeActionContextTypeResolver.java

```diff
@@ -94,6 +107,71 @@ public Optional<TypeSymbol> transform(BuiltinSimpleNameReferenceNode builtinSimp
         return SymbolUtil.getTypeDescriptor(symbol.orElse(null));
     }
 
+    public Optional<TypeSymbol> transform(FunctionDefinitionNode functionDefinitionNode) {
+        Optional<ReturnTypeDescriptorNode> returnTypeDesc = functionDefinitionNode.functionSignature().returnTypeDesc();
+        if (returnTypeDesc.isEmpty() || context.currentSemanticModel().isEmpty()) {
+            return Optional.empty();
+        }
+
+        Optional<Symbol> functionSymbol = context.currentSemanticModel().get().symbol(functionDefinitionNode);
+
+        if (functionSymbol.isEmpty()) {
+            return Optional.empty();
+        }
+
+        return ((FunctionSymbol) functionSymbol.get()).typeDescriptor().returnTypeDescriptor();
+    }
+
+    @Override
+    public Optional<TypeSymbol> transform(ConstantDeclarationNode constantDeclarationNode) {
+        if (context.currentSemanticModel().isEmpty()) {
+            return Optional.empty();
+        }
+        Optional<Symbol> symbol = context.currentSemanticModel().get().symbol(constantDeclarationNode);
```

Comment:
We can merge this logic as follows
```suggestion
        Optional<Symbol> symbol = context.currentSemanticModel().map(semanticModel-> semanticModel.symbol(constantDeclarationNode));
```